docs/pages/reference/architecture/architecture.mdx

```diff
@@ -1,10 +1,131 @@
+read about the architecture of Teleport Proxy Service features:
+
+- [TLS Routing](tls-routing.mdx)
+- [Proxy Peerint](proxy-peering.mdx)
+
+## Enrolling resources
+
+Administrators can **enroll** infrastructure resources with a Teleport cluster
+to provide secure access, RBAC, and auditing. There are three ways to enroll
+infrastructure resources with a Teleport cluster:
+
+- **Teleport Agents** proxy traffic from human users to and from
+  Teleport-protected infrastructure resources.
+- **Machine ID Bots** receive short-lived credentials from the `tbot` binary so
+  service accounts can access infrastructure.
+- **Trusted clusters** allow a user of one Teleport cluster to access
+  infrastructure that is enrolled with another Teleport cluster by federating
+  trust across multiple Teleport clusters.
+
+### Teleport Agents
+
+Teleport Agents proxy traffic from users to resources in your infrastructure.
+Agents are instances of the `teleport` binary configured to run data plane
+services, e.g., the Teleport SSH Service and Teleport Kubernetes Service, and
+administrators deploy Agents on their own infrastructure.
+
+Agents verify a user's certificate against a certificate authority maintained by
+the Teleport Auth Service. Since a user's Teleport roles are encoded in their
+certificate, a Teleport Agent can check a user's Teleport roles and permit or
+deny access to a resource. 
+
+Agents must establish trust with the Teleport Auth Service when first joining a
+cluster, and there is are [variety of
+methods](../enroll-resources/agents/join-services-to-your-cluster.mdx) that
+Agents use for this.
+
+Read more about [Teleport Agent Architecture](agents.mdx). You can also read
+about the architecture of Teleport Agent features:
+
+- [Automatic Agent updates](agent-update-management.mdx): How a
+  Teleport cluster ensures that Agents run the most up-to-date version of the
+  `teleport` binary.
+- [Automatically discovering Kubernetes
+  applications](docs/pages/reference/architecture//kubernetes-applications-architecture.mdx):
+  The Teleport Discovery Service queries your Kubernetes cluster and registers
+  applications with the Teleport Auth Service.
+- [Session recordings](session-recording.mdx): Teleport Agents record user
+  sessions and send the data to the Auth Service for storage.
+
+### Machine ID
+
+Machine ID is a Teleport system that enables automated services to access
+Teleport-protected infrastructure with regularly updated credentials.
+Administrators register a Bot user with Teleport that, like a human user, is
+assigned Teleport roles. 
+
+Instances of the `tbot` binary communicate with the Teleport Auth Service to
+continuously refresh credentials. As with Agents, administrators must deploy
+`tbot` on their own infrastructure, including on CI/CD platforms such as GitHub
+Actions. The [join methods](../enroll-resources/machine-id/deployment.mdx) for
+`tbot` differ from those for Agents.
```

Comment:
> The [join methods](../enroll-resources/machine-id/deployment.mdx) for
`tbot` differ from those for Agents.

The only join method not available to bots is the `ec2` join method since it was largely replaced by the `iam` join method.  All join methods availale for bots should also work for agents (we just don't tend to write guides for them as they don't make sense - e.g github actions).